### PR TITLE
Will not let stupid algorithm choose a move that is not legal.

### PR DESCRIPTION
**Pull request by k-owens**
Also create function to get available, legal moves for use in the algorithms.

### CODE REVIEW

File: TicTacToe/Game.fs [hunk 1]
```diff
@@ -24,4 +24,7 @@ let otherPlayer(game : Game, playerNumber) =
     if(playerNumber = 1) then
         game.Players.[1]
     else
-        game.Players.[0]
\ No newline at end of file
+        game.Players.[0]
+
+let possibleMoves (gameBoard) =
+    [for location in 0 .. (gameBoard.BoardSize*gameBoard.BoardSize - 1) -> if(gameBoard.CurrentBoard.[location] = None) then Some(location) else None] |> List.choose id
```
Comment (croese) [inline TicTacToe/Game.fs:30]: Like the refactoring here to get rid of duplicated code

File: TicTacToe.test/UnitTests.fs [hunk 2]
```diff
@@ -365,8 +365,8 @@ let canComputerFindDiagonalWins () =
...
 [<Fact>]
 let canStupidComputerMakeMove () =    
-    let board = {BoardSize = 3; CurrentBoard = [None;None;None;None;None;None;None;None;None]; TurnNumber = 1; IsInverted = false}
+    let board = {BoardSize = 3; CurrentBoard = [Some('O');Some('O');None;None;Some('X');None;None;None;None]; TurnNumber = 1; IsInverted = false}
     let player1 = {PlayerType = 1; ComputerAlgorithm = minimaxStarter; PlayerCharacter = 'O'}
     let player2 = {PlayerType = 1; ComputerAlgorithm = minimaxStarter; PlayerCharacter = 'X'}
     let testMove = stupidComputerMove(board,player1,player2) 
-    Assert.True(testMove > -1 && testMove < 9)
\ No newline at end of file
+    Assert.True(testMove > 1 && testMove < 9 && not(testMove = 4))
```
Comment (croese) [inline TicTacToe.test/UnitTests.fs:372]: I'm not sure how this added condition relates to the testing of `canStupidComputerMakeMove` - why is space 4 singled out, specifically?

Comment (k-owens) [inline TicTacToe.test/UnitTests.fs:372]: It made sense to make it test a particular board.  Before, I was just testing a blank board, but that did not prove that the changes that were made for this commit actually worked.  Now, the test board has the spaces 0, 1, and 4 already selected.  That is why it should not pick 4, and why the "testMove > -1" changed to "testMove > 1"

Comment (croese) [inline TicTacToe.test/UnitTests.fs:372]: Ah, I missed the `> 1` part. Makes sense now

**Comment by croese** [comment]
:+1:

**Pull request merged**